types/recompose/recompose-tests.tsx

```diff
@@ -412,3 +412,11 @@ function testLifecycle() {
         }
     })(component)
 }
+
+function testShallowEqual() {
+    const a = { prop: 1 }
+    const b = { prop: 1 }
+    // Generics are not needed inside this function, however they should flow through for outside
+    // inference.
+    shallowEqual(a, b)
+}
```

Comment:
Would you be willing to add the test case from the PR description to the test file?
It seems like it motivates the change a lot better than what you have here, and therefore would better defend against any future breaking changes.

Reply:
For that we would need to import RxJS’s `Observable` into the tests, and I'm not sure how to do that 🤔

Reply:
As discussed [here](https://github.com/DefinitelyTyped/DefinitelyTyped#i-notice-some-packages-having-a-packagejson-here), you can add a `package.json` to this package referencing RxJS, and then you'll be able to reference the RxJS types in `index.d.ts` and in the tests.

Reply:
That would mean any users of the types would also download the dependency though, which is a bit much given it's for a test?

Reply:
Oh... good point. In that case, I think you can fake it by doing something like this:
```
type MyOperatorFunction<T, U> = (source: T) => U;
declare function myDistinctUntilChanged<T>(compare?: (x: T, y: T) => boolean): (source: T) => T;
declare function myPipe<A, B>(op1: MyOperatorFunction<{ prop: number }, A>, op2: MyOperatorFunction<A,B>): B;
declare function myMap<T, R>(project: (value: T, index: number) => R, thisArg?: any): MyOperatorFunction<T, R>;
function testShallowEqual2(props$: { prop: number }) {
    myPipe(
       myDistinctUntilChanged(shallowEqual),
       myMap(props => props.prop) 
    );
}
```
You'll probably want to clean that up a bit, I'm not that familiar with what these calls actually represent.